File: app-cpp/src/ServerAppInterface.cpp
#include "ServerAppInterface.hpp"

#include <iostream>
#include <vector>
#include <string>

ServerAppInterface::ServerAppInterface(std::shared_ptr<ServerProcessingEngine> engine) : engine(engine) { }

void ServerAppInterface::readCommands() {
    // TO-DO implement the read commands method
    std::string command;
    
    while (true) {
        std::cout << "> ";
        
        // read from command line
        std::getline(std::cin, command);

        // if the command is quit, terminate the program       
        if (command == "quit") {
            engine->shutdown();
            break;
        }

        // if the command begins with list, list all the connected clients
        if (command.size() >= 4 && command.substr(0, 4) == "list") {
            // TO-DO call the getConnectedClients method from the server to retrieve the clients information
            // TO-DO print the clients information
            std::vector<std::string> clientsInformation = engine->getConnectedClients();
            continue;
        }

        std::cout << "unrecognized command!" << std::endl;
    }
}

File: app-java/src/main/java/csc435/app/ServerAppInterface.java
package csc435.app;

import java.lang.System;
import java.util.Scanner;
import java.util.ArrayList;

public class ServerAppInterface {
    private ServerProcessingEngine engine;

    public ServerAppInterface(ServerProcessingEngine engine) {
        this.engine = engine;
    }

    public void readCommands() {
        // TO-DO implement the read commands method
        Scanner sc = new Scanner(System.in);
        String command;
        
        while (true) {
            System.out.print("> ");
            
            // read from command line
            command = sc.nextLine();

            // if the command is quit, terminate the program       
            if (command.compareTo("quit") == 0) {
                engine.shutdown();
                break;
            }

            // if the command begins with list, list all the connected clients
            if (command.length() >= 4 && command.substring(0, 4).compareTo("list") == 0) {
                // TO-DO call the getConnectedClients method from the server to retrieve the clients information
                // TO-DO print the clients information
                continue;
            }

            System.out.println("unrecognized command!");
        }

        sc.close();
    }
}
